salt/utils/schedule.py

```diff
@@ -160,7 +160,18 @@ def __singleton_init__(
                 self.returners = returners
             else:
                 self.returners = returners.loader.gen_functions()
-        self.time_offset = self.functions.get("timezone.get_offset", lambda: "0000")()
+        try:
+            self.time_offset = self.functions.get(
+                "timezone.get_offset", lambda: "0000"
+            )()
+        except Exception:  # pylint: disable=W0703
```

Comment:
Do we know what Exception is raised here? Its usually better to be explicit about the exception we are catching.

Reply:
In the linked issue it was `pytz.exceptions.UnknownTimeZoneError`. But since scheduler is using `timezone.get_offset` (and if there is exception, minion will fail on start), it should be possible to get any exception here in theory (such as with wrong custom module)

From the core modules it could be:
 - pytz.exceptions.* from win_timezone
 - CommandExecutionError from mac_timezone
 - CommandExecutionError from timezone (or potentially lots of others, depending on the state of OS)

So I thought it would be good idea to ensure minion can start and function if there is any immediate timezone-related problem (although it violates W0703 rule). The second problem with catching pytz.exceptions.* directly might be whether it's really available on all supported platforms by default. It's hard dependency on Windows and it's in static deps for some other platforms, but it's not directly used outside win_timezone.

But I can change it to specific list of exceptions that might be relevant, if it's necessary.

Reply:
Thanks for the explanation. I'm good with your approach. :)